frontend/packages/kubevirt-plugin/src/components/vm-snapshots/vm-snapshots.tsx

```diff
@@ -0,0 +1,150 @@
+import * as React from 'react';
+import { Table, RowFunction } from '@console/internal/components/factory';
+import { sortable } from '@patternfly/react-table';
+import { getCreationTimestamp, getName, getNamespace, dimensifyHeader } from '@console/shared';
+import { useSafetyFirst } from '@console/internal/components/safety-first';
+import { Button } from '@patternfly/react-core';
+import { EmptyBox } from '@console/internal/components/utils';
+import {
+  WatchK8sResource,
+  useK8sWatchResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { getCondition } from '@console/knative-plugin/src/utils/condition-utils';
+import { getStatusConditions } from '../../selectors/selectors';
+import { getSnapshotVmName } from '../../selectors/snapshot/snapshot';
+import { VMSnapshot } from '../../types';
+import { isVMI } from '../../selectors/check-type';
+import { wrapWithProgress } from '../../utils/utils';
+import { VMLikeEntityTabProps } from '../vms/types';
+import { snapshotsTableColumnClasses } from './utils';
+import { ADD_SNAPSHOT, READY } from '../../utils/strings';
+import { VirtualMachineSnapshotModel } from '../../models';
+import { SnapshotRow } from './snapshot-row';
+import SnapshotModal from '../modals/snapshots-modal';
+import { asVM, isVMRunningOrExpectedRunning } from '../../selectors/vm';
+
+export type VMSnapshotsTableProps = {
+  data?: any[];
+  customData?: object;
+  row: RowFunction;
+  columnClasses: string[];
+  loadError: any;
+  loaded: boolean;
+};
+
+const NoDataEmptyMsg = () => <EmptyBox label="Snapshots" />;
+
+export const VMSnapshotsTable: React.FC<VMSnapshotsTableProps> = ({
+  data,
+  customData,
+  row: Row,
+  columnClasses,
+  loaded,
+  loadError,
+}) => {
+  const snapshotsRowData = data.map((snapshot) => ({
+    name: getName(snapshot),
+    namespace: getNamespace(snapshot),
+    date: getCreationTimestamp(snapshot),
+    status: getCondition(getStatusConditions(snapshot), READY)?.status,
```

Comment:
the status is more complex.
- There are two (Ready and Progressing) conditions which should be read and final status created.
- we should also account for the Error field and potentially ReadyToUse one as well

Reply:
it could be worth creating an enum for the final status used by UI